### ISSUE TITLE
Issues with training a classifier to identify individual frogs

### ISSUE THREAD

**Issue by kahst** [kahst/BirdNET-Analyzer#158]
I am trying to use the windows GUI version of BirdNET to train my own classifier in an attempt to identify individual frogs in my recordings.

The species is Allobates femoralis with a highly stereotypic call, where in previous studies colleagues and us have already shown that the calls are individually distinct, based on simple call parameters (spectral and temporal; most individual information is contained in the internote-intervalls of the 4-note call).

 

Gasser H, Amézquita Torres A, Hödl W (2009) Who is calling? Intraspecific call variation in the aromobatid frog Allobates femoralis. Ethology 115:596–607. doi: 10.1111/j.1439-0310.2009.01639.x

Tumulty JP, Pašukonis A, Ringler M, Forester JD, Hödl W, Bee MA (2018) Brilliant-thighed poison frogs do not use acoustic identity information to treat territorial neighbours as dear enemies. Anim Behav 141:203–220. doi: 10.1016/j.anbehav.2018.05.008

 

Before jumping to field recordings, I first want to investigate the general usefulness and ability of BirdNET to train on and subsequently classify these individual frog calls. To this end I am using labelled recordings of 25 individuals that were recorded each in their terrariums over several weeks. I have created a training dataset of 100 recordings (call bouts, consisting of multiple 4 note calls) that I cut out using the raven band energy detector. From these recordings, I have isolated a total of 11083 calls, coming from theses 25 individuals. Each call/training recording, consisting of 4 notes (upward sweeps ~2-4 kHz), is between 0.4-0.5 seconds long. The file format is 16 kHz mono, 16 bit WAV – resampled in Audacity from 48 kHz mono, 24 bit original recordings.

When trying to run the training function of the BirdNET GUI version with the default settings, the 25 labels/individuals are found in their folders, but training stops after a few epochs with the last messages:

 

Epoch 11/100

347/347 [==============================] - 2s 5ms/step - loss: 5.7809 - prec: 0.5430 - val_loss: 5.1439 - val_prec: 0.5214

WARNING:absl:Found untraced functions such as _update_step_xla while saving (showing 1 of 1). These functions will not be directly callable after loading.

...Done. Best top-1 precision: 0.5511953234672546

 

When changing the training parameters (epochs, learning rate, hidden units), the number of epochs until the process stops is varying slightly (9-17), but the final messages remains the same.

 

A *.tflite classifier file is getting written into the assigned folder and a precision plot is shown in the GUI. However, when I try to use this classifier to analyse my full recordings, I get the following error message for every single file (same audio format) to be analysed:

 

Error: Cannot analyze audio file D:\To_Classify\frog01_bout020882.wav.

Error: Cannot analyze audio file D:\To_Classify\frog01_bout020899.wav.

…

Etc.

 

To verify if the audio format is correctly processed, I tried to run the full BirdNET classifier with all species against my recordings. Then I receive no error messages and my frogs, all individuals, get classified as “Spring Peepers” (if I remember right).

 

Do you have any idea what might be the issue here that the training apparently stops prematurely and results in a broken classifier? It might be worth noting, that currently I am trying to run BirdNET on a rather outdated computer, a dual CPU XEON E5620 @ 2.4 GHz from 2009, which had issues with running CUDA (on an Nvidia GTX 2060) tensorflow for Koogu, as it lacks the ability to run certain needed, current CPU commandsets. However, as far as I understand BirdNET is not using CUDA, so this rather should not be a problem. Also the fact that the original BirdNET classifier works fine (although resulting in nonsensical identifications with my recordings for obvious reasons) rather tells me, that it is not about the hardware here.

Any help would be highly appreciated!

Kind Regards

Max

**Comment by kahst** [comment]
Max reached out to us via support email, I'm posting the issue here so that we can look into it.

**Comment by Josef-Haupt** [comment]
This has been stale for some time, please retry with the current main and create a new issue if necessary.
